dingo/gw/dataset/waveform_dataset.py

```diff
@@ -221,7 +235,21 @@ def __getitem__(self, idx) -> Dict[str, Dict[str, Union[float, np.ndarray]]]:
         data = {"parameters": parameters, "waveform": polarizations}
         if self.transform is not None:
             data = self.transform(data)
-        return data
+
+        # currently the data is of form [arr1[batch_size, ...], arr2[batch_size, ...], ...]
+        # repackage it to [[arr1[0, ...], arr2[0, ...], ] ..., [arr1[batch_size, ...], arr2[batch_size, ...], ]]
```

Comment:
The comment is a little confusing, because `arr1[batch_size, ...]` in the first line corresponds exactly to the element in the output `arr1[batch_size, ...]` (second line). This might give a wrong impression since the array in the first line has shape [batch_size, ...], while the array in the second line has shape [1, ...] since you are indexing into the full array.
Maybe it might help to say that `[batch_size, ...]` is the shape of `arr1` in the beginning, and `[1,...]` in the end?

One could also report the complete shape: 
```
# Currently, the data is of shape [M, batch_size, ...] with M the number of arrays returned by the transform.
# This array is repackaged to group different indices of `M` into one sample, resulting in data of shape [batch_size, M, ...].
```
Probably it would be best to combine the comment above with your current description of what is actually happening inside the repackaging (and give `batch_size` explicit indices, e.g. `0,...n,..., N` with `N = batch_size`)

Reply:
Good point, I read what I wrote earlier and was confused by it. I updated the docs, let me know if it's clearer now.